PhysicsTools/PatAlgos/plugins/PATTauProducer.cc

```diff
@@ -378,13 +375,43 @@ void PATTauProducer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
       // aTauPFEssential.pvPos_ = tauLifetimeInfo.primaryVertexPos();
       // aTauPFEssential.pvCov_ = tauLifetimeInfo.primaryVertexCov();
       aTauPFEssential.hasSV_ = tauLifetimeInfo.hasSecondaryVertex();
-      if(tauLifetimeInfo.hasSecondaryVertex()){
-	aTauPFEssential.flightLength_ = tauLifetimeInfo.flightLength();
-	aTauPFEssential.flightLengthSig_ = tauLifetimeInfo.flightLengthSig();
-	//      aTauPFEssential.sv_ = tauLifetimeInfo.secondaryVertex();
-	// aTauPFEssential.svPos_ = tauLifetimeInfo.secondaryVertexPos();
-	// aTauPFEssential.svCov_ = tauLifetimeInfo.secondaryVertexCov();
-      }
+      //if(tauLifetimeInfo.hasSecondaryVertex()){
+      aTauPFEssential.flightLength_ = tauLifetimeInfo.flightLength();
+      aTauPFEssential.flightLengthSig_ = tauLifetimeInfo.flightLengthSig();
+      //      aTauPFEssential.sv_ = tauLifetimeInfo.secondaryVertex();
+      // aTauPFEssential.svPos_ = tauLifetimeInfo.secondaryVertexPos();
+      // aTauPFEssential.svCov_ = tauLifetimeInfo.secondaryVertexCov();
+      //}
+      aTauPFEssential.ip3d_ = tauLifetimeInfo.ip3d();
+      aTauPFEssential.ip3d_error_ = tauLifetimeInfo.ip3d_error();
+    }
+
+    // extraction of variables needed to rerun MVA isolation on MiniAOD
+    // (only available for PFTaus)
+    if( aTau.isPFTau() ) {
+        edm::Handle<reco::PFTauCollection> pfTaus;
+        iEvent.getByToken(pfTauToken_, pfTaus);
+        reco::PFTauRef pfTauRef(pfTaus, idx);
+        pat::tau::TauPFEssential& aTauPFEssential = aTau.pfEssential_[0];
+        float ecalEnergy = 0;
+        float hcalEnergy = 0;
+        std::vector<reco::PFCandidatePtr> signalCands = pfTauRef->signalPFCands();
```

Comment:
why is a copy of signalCands needed?
The current implementation suggests ``` auto const& signalCands = pfTauRef->signalPFCands();``` will work.
Also, a "for range loop" can work to make the code less verbose